test/streamer/asyncOpStream.test.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { expect, test } from '@oclif/test';
```

Comment:
I think that here we should test the stream functionality only, in isolation. We shouldn't have to execute the deploy pipeline nor stub the fetchAndValidatePipelineStage() method.
Then in te test for deploy:pipeline we can stub the stream service and assert it was called.